### DISCUSSION TITLE
UX: Split web search and file upload into two seperate buttons

### DISCUSSION THREAD

**Discussion by robertosw** [open-webui/open-webui#4918] [Ideas]
![image](https://github.com/user-attachments/assets/bdb182ef-5e53-4644-93cf-d997d61d2f7d)

I frequently turn the web search on and off. Mostly because I have a big question which the llm should search about and then I have some smaller questions. If I enable web search for the smaller follow-up questions, the llms dont include the context of the conversation into the search, so the answer is quite useless. Disabling the web search leads to much better answers.

Because of this, I find it quite annoying that this toggle is behind another click.

## Proposal

Web Search disabled:
![image](https://github.com/user-attachments/assets/f81f8ee4-5101-43a8-afc4-023582268655)

Web Search enabled:
![image](https://github.com/user-attachments/assets/2565611d-0384-4d62-9f3f-18c1823c5b4a)

Clicking on the plus icon would have the same effect as clicking on "upload files" currently

**Comment by robertosw** [comment]
Saving the latest state of "web search on / off"  would resolve the topic of [this discussion](https://github.com/open-webui/open-webui/discussions/3390) as well

**Comment by MicAnKa** [comment]
This is really nice! I am on board!
